Install coreos VM on older qemu using the coreos iso

flyemsafe opened the pull request:
This addresses issue #44 .

I am running RHEL 7.6 kernel (3.10.0-957.el7.x86_64)

My host was already setup for libvirt. I added some tags to be able to skip tasks such as running yum update and installing packages. I also added **use_coreos_iso: true** to some task.

- add task to prepare-host.yml

This downloads my role swygue.coreos-virt-install-iso which does most of the work.
 * https://github.com/flyemsafe/swygue.coreos-virt-install-iso

- add task to create.yml

This calls the main.yml from the swygue.coreos-virt-install-iso. This roles basically downloads the coreos iso and the metal file. Setup up httpd by podman, and expose the ignition files and metal file. I also added **vm_role**.

- add task to create-vm.yml

This calls deploy_vm.yml swygue.coreos-virt-install-iso when **user_coreos_iso** is true.

tahonen commented:
I ll run this today on RHEL 7.6 and test.

tahonen commented:
More testing is needed on "vanilla" RHEL
* SELinux prevents bootstrap host from starting
* prepare host playbook tryis to install centos openstack release repos. Need host os check to enable correct repos. Openstack repos are needs fro qemu-kvn 2.4+
* python-boto package not found.
* Creating dirs for downloaded content and ignition didn't work. Had to create by hand.

flyemsafe commented:
* Can you expand on why Selinux is preventing the bootstrap host from starting. Are you talking about the bootstrap VM?

* I don't run the prepare host playbook as is, because I already have a role that configures by host. And I'm not doing anything with AWS to need the boto package. I also added a tags to skip the OpenStack repo, the tag is qemu_update. And added kvm_pkgs tag to skip the kvm packages. With all that said, you will need to register your rhel VM to RHSM and enable the right repos so you can install python-boto.

* Creating does did not work. You are probably don't have the variables declared.

rbo commented on ansible/roles/openshift-4-cluster/tasks/create-ignition.yml:
```diff
@@ -15,6 +15,7 @@
 
 - name: Create ignition files
-  command: "openshift-install --dir={{ openshift_install_dir }} create ignition-configs"
+  command: "/usr/local/bin/openshift-install --dir={{ openshift_install_dir }} create ignition-configs"
```
Why `/usr/local/bin/` ? It should be something like `/opt/openshift-install-4.2.0-0.nightly-2019-08-29-062233/openshift-install` ?

rbo replied:
Indeed we should use everywhere the full path from `/opt/openshift-install-{{ openshift_version }}/`
Opened an issue #47

flyemsafe replied:
I added /usr/local/bin because that is where it's symlink to.

```
udo ls -lasF /usr/local/bin/openshift-install 
0 lrwxrwxrwx. 1 root root 74 Sep 29 16:07 /usr/local/bin/openshift-install -> /opt/openshift-install-4.2.0-0.nightly-2019-09-24-194016/openshift-install*
```

I figure that's better rather that always keeping up with the **openshift_version**.

rbo replied:
It depends, I have several different cluster versions on my box. Creating the symlink make sense for user interaction, but the playbooks should use the exact version configured in the cluster.yml.

rbo replied:
I created the symlinks in https://github.com/RedHat-EMEA-SSA-Team/hetzner-ocp4/blame/master/ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml haha - totaly forgot that..  :-D

rbo commented on ansible/roles/openshift-4-cluster/tasks/create-vm.yml:
```diff
@@ -22,8 +31,27 @@
+
+- name: deploying vm instance using coreos iso
+  import_role:
+    name: swygue.coreos-virt-install-iso
```
Why your own ISO and not the official one from [rhcos-42.80.20190828.2-installer.iso](https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/pre-release/latest/rhcos-42.80.20190828.2-installer.iso) ?

flyemsafe replied:
Did you take a look at the code? I am downloading the `rhcos-42.80.20190828.2-installer.iso`. That's just the name of the role i use to prep to get the host ready for an install using the iso vs using the qcow image.

rbo replied:
I can not fine the role, please add the downloading to [download-openshift-artifacts.yml](/RedHat-EMEA-SSA-Team/hetzner-ocp4//blob/master/ansible/roles/openshift-4-cluster/tasks/download-openshift-artifacts.yml)

rbo commented on ansible/roles/openshift-4-loadbalancer/tasks/destroy.yml:
```diff
@@ -11,5 +11,22 @@
+  when: running_containers.stdout != ""
+  with_items: "{{ running_containers.stdout_lines }}"
+  become: yes
+
```
Good point, we should handle the cleanup of the container at the service stop task.

rbo commented on inventory/inventory.3.11.rhel.gluster:
```diff
@@ -0,0 +1,95 @@
+openshift_metrics_cassandra_nodeselector={"node-role.kubernetes.io/infra":"true"}
+openshift_metrics_heapster_nodeselector={"node-role.kubernetes.io/infra":"true"}
+
+
```
Why we need an OpenShift 3.11 inventory?

flyemsafe replied:
Oops! we don't need it. I copied the inventory folder and did not do a cleanup. I will create an issue.

tahonen commented:
About that SELinux...I had no time to look about it in any dept. Bootstrap VM was not starting libvirt error. Disabled SELinux and it started working. :D

tahonen commented:
python-boto is needed if route53 DNS is used, otherwise unnecessary.

tahonen commented:
If there is some specific repos that are needed, those need to be enabled by playbooks and when RHEL also subscribing the host needs to be taking care of. Meaning that RHN user and pwd + pool id needs to be taken care of.

tahonen commented:
Cannot run setup.yml playbook without adding swygue.coreos-virt-install-iso role manually under roles.

flyemsafe commented:
> Cannot run setup.yml playbook without adding swygue.coreos-virt-install-iso role manually under roles.

What does this mean? The role gets deployed by the task prepare-host.yml.

```
- name: download swygue.coreos-virt-install-iso
  git:
    repo: "{{ virt_install_iso_role_url }}"
    dest: "roles/{{ virt_install_iso_role }}"
  when: use_coreos_iso|bool
```

tahonen commented:
That was odd since I saw that task in there. Added role manually and then it worked....i ll grep that error tomorrow on online

rbo commented:
Is this still being worked on? Should we leave it open ?!

tahonen commented:
No glue

rbo commented:
closed due to long inactivity. Feel free to reopen...
